### PR TITLE
docs: suggest comment improvements and update pragma

### PR DESCRIPTION
**Pull request by eccentricexit**
Why do we require transactions to be ordered by bigger hash? Couldn't this cause problems if one transaction requires the execution of a tx before it?

Why do we set the meta evidence in another step instead of doing it in the constructor?

L487-L494: I think this was answered before and probably warrants a comment on the code. Why not use check-effects-interaction? E.g.
```
...
require(!transaction.executed && transaction.value <= expendableFunds!transaction.executed && transaction.value <= expendableFunds);
transaction.executed = true;
transaction.target.call...
...
```

### CODE REVIEW

File: contracts/kleros/KlerosGovernor.sol [hunk 1]
```diff
@@ -104,16 +104,16 @@ contract KlerosGovernor is Arbitrable {
      *  @param _description The string in CSV format that contains labels of list's transactions.
      *  Note that the submitter may give bad descriptions of correct actions, but this is to be seen as UI enhancement, not a critical feature and that would play against him in case of dispute.
      */
-    event ListSubmitted(uint indexed _listID, address indexed _submitter, uint _session, string _description);
+    event ListSubmitted(uint indexed _listID, address indexed _submitter, uint indexed _session, string _description);
 
     /** @dev Constructor.
-     *  @param _arbitrator The arbitrator of the contract. It should support appealPeriod.
+     *  @param _arbitrator The arbitrator of the contract. It must support appealPeriod.
```
Comment (mtsalenc) [inline contracts/kleros/KlerosGovernor.sol:110]: We should be clearer on this. Either it is required or not.

**Comment by unknownunknown1** [comment]
> Why do we require transactions to be ordered by bigger hash?

It is to prevent duplicates, when both lists have the same transactions but ordered differently.

> Couldn't this cause problems if one transaction requires the execution of a tx before it?

That's a good question. Perhaps Clement will answer this, since this requirement was there from the start

> Why do we set the meta evidence in another step instead of doing it in the constructor?

It was in the contstructor initially, but Sam changed it. I'll ask why

> L487-L494: I think this was answered before and probably warrants a comment on the code. Why not use check-effects-interaction? E.g

I believe falied external call doesn't revert so we need to check that it was actually successful before setting `executed = true.`

File: contracts/kleros/KlerosGovernor.sol [hunk 2]
```diff
@@ -104,16 +104,16 @@ contract KlerosGovernor is Arbitrable {
...
      *  @param _extraData Extra data for the arbitrator.
      *  @param _submissionBaseDeposit The base deposit required for submission.
      *  @param _submissionTimeout Time in seconds allocated for submitting transaction list.
      *  @param _executionTimeout Time in seconds after approval that allows to execute transactions of the approved list.
      *  @param _withdrawTimeout Time in seconds after submission that allows to withdraw submitted list.
-     *  @param _sharedMultiplier Multiplier of the appeal cost that submitters has to pay for a round when there is no winner/loser in the previous round. In basis points.
+     *  @param _sharedMultiplier Multiplier of the appeal cost that submitters has to pay for a round when there is no winner/loser in the previous round (e.g. when there isn't a previous round). In basis points.
```
Comment (unknownunknown1) [inline contracts/kleros/KlerosGovernor.sol:116]: I think this wording makes it more confusing. I'd rather say "e.g when it's the very first round"

File: contracts/kleros/KlerosGovernor.sol (hunk 1 above)
Comment (unknownunknown1) [reply contracts/kleros/KlerosGovernor.sol:110]: These words are interchangeable to me but "must" would be more gramatically correct, yeah

File: contracts/kleros/KlerosGovernor.sol [hunk 3]
```diff
@@ -171,7 +171,7 @@ contract KlerosGovernor is Arbitrable {
         executionTimeout = _executionTimeout;
     }
 
-    /** @dev Changes the time allowed for list withdrawal.
+    /** @dev Changes list withdrawal timeout. Note that the withdrawal period is the intersecion between the withdraw timeout and the first half of the submission period.
```
Comment (unknownunknown1) [inline contracts/kleros/KlerosGovernor.sol:174]: This makes it sound more complicated than it actually is. I'd rather not add anything or at least rephrase it to "Note that withdrawals are only possible in the first half of the submission period"

Also, "intersecion" -> "intersection"

**Review by unknownunknown1 — APPROVED** [review]
